kaali001/varta
# Implement Custom Permission Popover for Camera and Audio Access

## Description

This update introduces a custom permission popover to request user permission for camera and audio access. The component is designed to streamline the user experience by managing permissions within the active session and minimizing redundant prompts. Initially , I thought it would take me a couple hours but I was wrong. It became more complicated but at the end it worked out fine. And sorry , I had to PR with my master branch as I was too focus on developing it.

Fixes #21 


## Changes

List the changes done to fix a bug or introducing a new feature.

I had to apply many chances and introduce new states,effects that hadles various situations. There were few challenges while creating this component. 

- Introduced multiple components to manage and display the user’s current permission status, along with prompts to request permissions as needed.
- Ensured cross-browser compatibility for consistent behavior across different environments.
- Implemented multiple useEffect hooks and state variables to handle various permission and event states reliably.
- The if conditions are a bit to hard to explain without actually running the program. So, I have left multiple comments. 


## How to test

Simply open the development server and hit the join button at https://localhost:3000/chat, you will notice a custom permission component.




## Checklist:

- [Y] My code follows the style guidelines of this project
- [Y] I have performed a self-review of my code
- [Y] I have commented my code, particularly in hard-to-understand areas
- [N] I have made corresponding changes to the documentation
- [Y] My changes generate no new warnings
- [Y] I have added tests that prove my fix is effective or that my feature works
